server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go

```diff
@@ -0,0 +1,304 @@
+	req, err := procedure.GetRequestFromEvent[*callbackRequest](event)
+	if err != nil {
+		procedure.CancelEventWithLog(event, err, "get request from event")
+		return
+	}
+	params := req.p.params
+
+	// Select first shard to create partition table.
+	partTableShardNode := params.PartitionTableShards[0]
+
+	partInfo := req.p.params.SourceReq.GetPartitionTableInfo().GetPartitionInfo()
+	createTableResult, err := procedure.CreateTableMetadata(req.ctx, params.ClusterMetadata, params.SourceReq.GetSchemaName(), params.SourceReq.GetName(), partTableShardNode.ShardInfo.ID, partInfo)
+	if err != nil {
+		procedure.CancelEventWithLog(event, err, "create table metadata")
+		return
+	}
+	req.p.createTableResult = createTableResult
+
+	if err = procedure.CreateTableOnShard(req.ctx, params.ClusterMetadata, params.Dispatch, partTableShardNode.ShardInfo.ID, procedure.BuildCreateTableRequest(createTableResult, params.SourceReq, partInfo)); err != nil {
+		procedure.CancelEventWithLog(event, err, "dispatch create table on shard")
+		return
+	}
+}
+
+// 2. Create data tables in target nodes.
+func createDataTablesCallback(event *fsm.Event) {
+	req, err := procedure.GetRequestFromEvent[*callbackRequest](event)
+	if err != nil {
+		procedure.CancelEventWithLog(event, err, "get request from event")
+		return
+	}
+	params := req.p.params
+
+	for i, subTableShard := range params.SubTablesShards {
+		createTableResult, err := procedure.CreateTableMetadata(req.ctx, params.ClusterMetadata, params.SourceReq.GetSchemaName(), params.SourceReq.GetPartitionTableInfo().SubTableNames[i], subTableShard.ShardInfo.ID, nil)
+		if err != nil {
+			procedure.CancelEventWithLog(event, err, "create table metadata")
+			return
+		}
+
+		if err = procedure.CreateTableOnShard(req.ctx, params.ClusterMetadata, params.Dispatch, subTableShard.ShardInfo.ID, procedure.BuildCreateTableRequest(createTableResult, params.SourceReq, nil)); err != nil {
+			procedure.CancelEventWithLog(event, err, "dispatch create table on shard")
+			return
+		}
+	}
+}
+
+func finishCallback(event *fsm.Event) {
+	req, err := procedure.GetRequestFromEvent[*callbackRequest](event)
+	if err != nil {
+		procedure.CancelEventWithLog(event, err, "get request from event")
+		return
+	}
+	log.Info("create partition table finish")
+
+	if err := req.p.params.OnSucceeded(req.p.createTableResult); err != nil {
+		procedure.CancelEventWithLog(event, err, "create partition table on succeeded")
+		return
+	}
+}
+
+func (p *Procedure) updateStateWithLock(state procedure.State) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	p.state = state
+}
+
+func (p *Procedure) persist(ctx context.Context) error {
+	meta, err := p.convertToMeta()
+	if err != nil {
+		return errors.WithMessage(err, "convert to meta")
+	}
+	err = p.params.Storage.CreateOrUpdate(ctx, meta)
+	if err != nil {
+		return errors.WithMessage(err, "createOrUpdate procedure storage")
+	}
+	return nil
+}
+
+type rawData struct {
```

Comment:
I guess we should not define the format for persisting here, which may lead to compatibility problems. How about define it in the ceresdbproto?